src/unix/aix.c

```diff
@@ -329,25 +331,26 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
     if (have_signals != 0) {
       uv__metrics_update_idle_time(loop);
       loop->signal_io_watcher.cb(loop, &loop->signal_io_watcher, POLLIN);
+      did_work = 1;
     }
 
     loop->watchers[loop->nwatchers] = NULL;
     loop->watchers[loop->nwatchers + 1] = NULL;
 
     if (have_signals != 0)
-      return;  /* Event loop should cycle now so don't poll again. */
+      return did_work;  /* Event loop should cycle now so don't poll again. */
```

Comment:
It should be enough to return whether `nevents > 0`, and simply `return 0` in every location above this.

Should be pretty much the same wherever `nevents` is used in other locations.